pkg/reconciler/common/deployment.go

```diff
@@ -21,11 +21,26 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// IsDeploymentAvailable returns the whether the deployment is ready for the revision to be scaled up.
 func IsDeploymentAvailable(d *appsv1.Deployment) bool {
+	if d.Spec.Replicas == nil || *d.Spec.Replicas == 0 || d.Status.AvailableReplicas != *d.Spec.Replicas {
```

Comment:
availableReplica may not necessarily be equal to the replicas as you can specify `maxUnavailable` in k8s, I am not sure if this is true for knative.

Reply:
Knatve Serving sets maxUnavailable to 0.